source/agora/consensus/data/UTXOSet.d

```diff
@@ -81,25 +126,116 @@ public class UTXOSet
 
         Params:
             tx = the transaction
+            height = Height of the block where UTXO will be activated
 
     ***************************************************************************/
 
-    public void updateUTXOCache (const ref Transaction tx) @safe
+    public void updateUTXOCache (const ref Transaction tx, ulong height) @safe
     {
-        foreach (const ref input; tx.inputs)
+        if (tx.type == TxType.Freeze)
         {
-            auto utxo_hash = getHash(input.previous, input.index);
-            this.utxo_db.remove(utxo_hash);
+            UTXOSetValue utxo_value;
+            foreach (const ref input; tx.inputs)
+            {
+                if (!this.getUTXOSetValue(input.previous, input.index, utxo_value))
+                    return;
+                else if (utxo_value.type != TxType.Payment)
+                    return;
+            }
+
+            foreach (const ref input; tx.inputs)
+            {
+                auto utxo_hash = getHash(input.previous, input.index);
+                this.utxo_db.remove(utxo_hash);
+            }
+
+            ulong actived_height = height + 1;
```

Comment:
Why +1? The UTXO will be in the block at the provided height that's in the parameter, not height + 1.

Reply:
Next Block height

Reply:
I entered the available height, i.e. unlock height.
I wanted to specify what could be used in the next block, not in the current block.